### PR TITLE
User story7

### PR DESCRIPTION
**Pull request by anotherMichaelMaier**

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+          <button onClick={moveTaskToPreviousState}>{previousState}</button>
+        )}
+        <button onClick={handleClick}>
+          {toggleDetailsCard ? "show details" : "hide details"}
+        </button>
+        {nextState && (
+          <button onClick={moveTaskToNextState}>{nextState}</button>
+        )}
+      </div>
+      {toggleDetailsCard ? null : (
```
Comment (Long-HaHoang) [inline components/Card.js:32]: Well in the end you implemented the previousState and nextState.
Looks good 👍

File: components/FilteredList.js [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
-          />
-        </StyledLi>
-      ))}
+      {filteredTasks.map((mappedTask) => {
+        const possibleStates = ["todo", "doing", "done"];
+        const indexOfCurrentState = possibleStates.findIndex(
+          (possibleState) => possibleState === mappedTask.position
+        );
+        const previousState = possibleStates[indexOfCurrentState - 1];
+        const nextState = possibleStates[indexOfCurrentState + 1];
```
Comment (Long-HaHoang) [inline components/FilteredList.js:14]: Nice one. Enumeration is a nice solution. 👍

File: pages/_app.js [hunk 3]
```diff
@@ -19,14 +19,31 @@ export default function App({ Component, pageProps }) {
...
+  function updateTask(taskId, updatedTask) {
+    setTasks((oldTasks) => {
+      const newTasks = oldTasks.map((oldTask) => {
+        if (oldTask.id === taskId) {
+          return updatedTask;
+        } else {
+          return oldTask;
+        }
+      });
+      return newTasks;
```
Comment (Long-HaHoang) [inline pages/_app.js:31]: Well, it's an extra step to save it to a const before returning it.
But works fine and it's readable.

**Review by Long-HaHoang — APPROVED** [review]
Looks good to me.
This solution works great.

File: pages/_app.js (hunk 3 above)
Comment (anotherMichaelMaier) [reply pages/_app.js:31]: Exactly, readability was important for me here.

**Pull request merged**